### PR TITLE
Adds a button to player panel that allows for admins to easily re-create a playermob (while providing debug information for later)

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Adds a button to player panel, "recreate mob". This button will delete the player's existing mob and then re-create a duplicate instance of it at its place, passing over equipment / held items / implants / quirks. 
- This button has an optional alert for the admin using it to create a debug dump file. Doing so will dump all of the mob's vars into a json so that an investigation into why the mob was troublesome can follow later. (Suggestion by MSO)

## Why It's Good For The Game

- Somewhat over, a player manages to screw up their mob so bad that most admins not experienced with the code has one option: Just delete the guy and respawn them. This provides a much easier avenue for them to do that, keeping clothing and equipment, as well as some other important facets like quirks and implants. 
- At the same time, if there genuinely is an issue with the mob, this allows a snapshot of the mob's vars to be saved, so that a coder can, later, look through it to see what the issue was. 

## Changelog

:cl: Melbert
admin: Added a new button to the player panel, "Recreate Mob". Pressing it will (after a confirmation box) delete the selected player's mob and duplicate it in their place, keeping their equipment / quirks / implants mostly intact. This is designed to be used for cases in which a player's mob is FUBAR and you must respawn them (NOT for a ghetto aheal). The button comes with the ability to create a dump file for you to hand off to coders after the round, so they can investigate what was wrong.
/:cl:

### CODE REVIEW

File: code/modules/admin/topic.dm [hunk 1]
```diff
@@ -203,6 +203,107 @@
...
+
+		// Pass the mind over
+		if(old_mob.mind)
+			old_mob.mind.transfer_to(new_mob, TRUE)
+		else
+			new_mob.key = old_mob.key
+
+		// clean up the old mob in a tick
+		// we probably don't need to wait a tick but you never know
+		QDEL_IN(old_mob, 1)
```
Comment (san7890) [inline code/modules/admin/topic.dm:305]:
```suggestion
		// clean up the old mob in a tick
		// we probably don't need to wait a tick but you never know
		QDEL_IN(old_mob, 1 TICS)
```

units are good

**Comment by san7890** [comment]
Did you account for skillchips?

**Comment by san7890** [comment]
> > Did you account for skillchips?
> 
> Nope, I'll add in skillchips - that being said, I don't really want to chase all the edge cases to the end of the Earth, as in general it's not intended to be perfect

Yeah, I just noticed you were doing a bunch of stuff for implants and the like and wasn't sure if you included functionality for that attribute

**Comment by Cyprex** [comment]
Noooooooo I would have already gotten several uses out of this...

**Pull request closed without merging**